src/Util/Util.php

```diff
@@ -183,7 +183,7 @@ public static function extractErrorMessage( ?ResponseInterface $resp, RequestInt
 	public static function buildDOMDocumentFromHtml( string $html ): DOMDocument {
 		$document = new DOMDocument( '1.0', 'UTF-8' );
 		libxml_use_internal_errors( true );
-		$document->loadHTML( mb_convert_encoding( str_replace( '<?xml version="1.0" encoding="UTF-8" ?>', '', $html ), 'HTML-ENTITIES', 'UTF-8' ) );
+		$document->loadHTML( str_replace( '<?xml version="1.0" encoding="UTF-8" ?>', '', $html ), LIBXML_HTML_NODEFDTD );
```

Comment:
Out of curiosity question: I have seen you tried using the parser from `wikimedia/html-formatter`. Why is it not working?

Reply:
Yeah, I tried that, but it produces HTML rather than XHTML so validation kept failing for various reasons. :( I might have been doing something wrong though!

Reply:
Thank you!